server/src/main/java/org/elasticsearch/action/search/TransportOpenPointInTimeAction.java

```diff
@@ -213,6 +213,30 @@ SearchPhase openPointInTimePhase(
                 searchRequest.getMaxConcurrentShardRequests(),
                 clusters
             ) {
+                @Override
+                protected void doCheckNoMissingShards(
+                    String phaseName,
+                    SearchRequest request,
+                    GroupShardsIterator<SearchShardIterator> shardsIts
+                ) {
+                    final StringBuilder missingShards = new StringBuilder();
+                    // Fail-fast verification of all shards being available
+                    for (int index = 0; index < shardsIts.size(); index++) {
+                        final SearchShardIterator shardRoutings = shardsIts.get(index);
+                        if (shardRoutings.size() == 0) {
+                            if (missingShards.isEmpty() == false) {
+                                missingShards.append(", ");
+                            }
+                            missingShards.append(shardRoutings.shardId());
+                        }
+                    }
+                    if (missingShards.isEmpty() == false) {
+                        // Status red - shard is missing all copies and would produce partial results for an index search
+                        final String msg = "Cannot execute [open_point_in_time] action due to missing shards [" + missingShards + "].";
+                        throw new SearchPhaseExecutionException(phaseName, msg, null, ShardSearchFailure.EMPTY_ARRAY);
+                    }
+                }
```

Comment:
woah, I am unsure about this. This is a fairly big change in behavior for PIT isn't it?

Reply:
Tbh I think that the change is not that big since we're overriding the (previously static)`SearchPhase#doCheckNoMissingShards` with just a couple of changes:
* we don't have the `if (request.allowPartialSearchResults() == false) ` check anymore as this is explicitly set by the PIT action (in `TransportOpenPointInTimeAction#doExecute`)
* we have a more specific error message, w/o suggesting the `allow_partial_search_results` as a solution anymore

The only reason for this addition was to update the error message, so if you think that this isn't needed right now (as this is something that we plan to revisit soon-ish) I can omit this from the PR.

Reply:
@pmpailis I would prefer this change be in a separate, focused PR. This way foundations, et. al. can focus their review on it.

Reply:
Opened https://github.com/elastic/elasticsearch/pull/111320